scripts/utils.js

```diff
@@ -25,11 +25,21 @@ function wmAvailabilityCheck(url, onsuccess, onfail) {
  * @param url {string}
  * @return {bool}
  */
-function isValidSnapshotUrl(url) {
+function isValidUrl(url) {
   return ((typeof url) === "string" &&
     (url.indexOf("http://") === 0 || url.indexOf("https://") === 0));
 }
 
+// Function to check whether it is a valid URL or not
+function isNotExcludedUrl(url, blacklist) {
+  for (var i = 0; i < blacklist.length; i++) {
```

Comment:
Since we call this function too often, its better to do the following to optimise it:
```
for (var i = 0, len = blacklist.length; i < len; i++) {
```
This way, we avoid calculating the length of `blacklist` for every loop

Reply:
Would it be better to use a set for the excluded urls instead of a list, and then avoid using a loop entirely?

Reply:
But you are using `url.startsWith`, you are not doing exact matching.
The target URL could be http://localhost/something-anything

`set` would be good for exact matching.